### PR TITLE
Document node problem behaviour

### PR DESCRIPTION
**Pull request by rmohr**
Document how KubeVirt behaves in case issues on the node appear.

### CODE REVIEW

File: workloads/virtual-machines/unresponsive-nodes.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+If the whole node is unresponsive, deleting a `VirtualMachineInstance` via
+`kubectl delete vmi <myvmi>` alone will never remove the
+`VirtualMachineInstance`. In this case all pods on the unresponsive node need
+to be force-deleted: First make sure that the node is really dead. Then delete
+all pods on the node via a force-delete: `kubectl delete pod --force
+--grace-period=0 <mypod>`.
+
+As soon as the pod disappears and the heartbeat from virt-handler timed out,
+the VMIs will be moved to `Failed` state. If they were already marked for
+deletion they will simply disappear not. If not, they can be deleted now and
```
Comment (slintes) [inline workloads/virtual-machines/unresponsive-nodes.md]: there is a "not" too much after disappear

File: workloads/virtual-machines/unresponsive-nodes.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
+# Detecting and resolving Node issues
+
+KubeVirt has its own node daemon, called `virt-handler`. In addition to the
```
Comment (stu-gott) [inline workloads/virtual-machines/unresponsive-nodes.md]: Suggest removing first-person pronouns (our, us):

KubeVirt has its own node daemon, called `virt-handler`. In addition to the usual k8s methods of detecting issues on nodes, the `virt-handler` daemon has its own heartbeat mechanism. This allows for fine-tuned error handling of `VirtualMachineInstances`.

File: workloads/virtual-machines/unresponsive-nodes.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  nodeSelector:
+    kubevirt.io/schedulable: "true"
+  [...]
+```
+
+In case there is a communication issue or the host goes down, `virt-handler`
+can't update its labels and annotations any-more. Once the last
+`kubevirt.io/heartbeat` timestamp is older than five minutes, the KubeVirt
+node-controller kicks in and sets the `kubevirt.io/schedulable` label to
+`false`. As a consequence no VMIs will be scheduled any-more to this node, until
```
Comment (stu-gott) [inline workloads/virtual-machines/unresponsive-nodes.md]: Suggest rewording to:

As a consequence no more VMIs will be schedule to this node until `virt-handler` is connected again.

File: workloads/virtual-machines/unresponsive-nodes.md (hunk 1 above)
Comment (stu-gott) [inline workloads/virtual-machines/unresponsive-nodes.md]: Suggest rewording the final sentence:

If not, they can be deleted and will disappear almost immediately.

File: workloads/virtual-machines/unresponsive-nodes.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+As soon as the pod disappears and the heartbeat from virt-handler timed out,
+the VMIs will be moved to `Failed` state. If they were already marked for
+deletion they will simply disappear not. If not, they can be deleted now and
+should almost immediately disappear.
+
+## Timing considerations
+
+It takes up to five minutes until the KubeVirt cluster components can detect
+that virt-handler is unhealthy. During that time-frame it is possible that new
+VMIs are scheduled to such a node. If virt-handler is not capable of connecting
```
Comment (stu-gott) [inline workloads/virtual-machines/unresponsive-nodes.md]: suggest "to the affected node" instead of "to such a node"
